conf/test_panelprep.config

```diff
@@ -20,9 +20,10 @@ params {
     max_time   = '1.h'
 
     // Genome references
-    fasta  = params.pipelines_testdata_base_path + "reference_genome/21_22/hs38DH.chr21_22.fa"
-    panel  = "${projectDir}/tests/csv/panel.csv"
-    phased = true
+    fasta        = params.pipelines_testdata_base_path + "reference_genome/21_22/hs38DH.chr21_22.fa"
+    input_region = "${projectDir}/tests/csv/region.csv"
```

Comment:
True, the input_region could be `all` as it therefore extract all the regions from the fasta files.
I've modified it for the test.

Reply:
I was thinking that when `params.input_region` is `null`, the region should default to the full chromosome. To ensure our implementation handles this correctly, it would be beneficial to add a test case where `input_region` is not specified. This way, we can confirm that the code runs as expected for full chromosomes.

Reply:
Okay I'll try !

Reply:
So, it cannot be done as the fasta file contains the whole chr21 and chr22.
Therefore the regions are chr21:0-46709983 and chr22:0-50818468.
For a subset of test the fasta files would need to be of the regions of interest.

But if we subset the fasta file to a subregion using `samtools faidx` then all the tools using the fasta file will crash as the string name of the contigs will be `chr22:16570000-16610000` instead of `chr22`. I've tried with the `params.pipelines_testdata_base_path + "reference_genome/21_22/hs38DH.chr21_22.s.fa"`

There is no good solution from my point of view.
What I would suggest is:
- if you don't want to specify a region, then it will run at least on full contigs names provide by fasta (this cannot be done with the test datasets we have for the moment, but this should still work).
- if you want to run on a specific region, fill the csv region file.

Reply:
I understand, it is something we'll have to take this into account. We can add "medium-sized" test datasets (e.g. the ref panel for a single full chromosome) to inspect these edge cases before we run the full-size test. We use a subset of the full-size test, such as:

```
panel,chr,vcf,index
1000GP.s.norel,chr22,http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20201028_3202_phased/CCDG_14151_B01_GRM_WGS_2020-08-05_chr22.filtered.shapeit2-duohmm-phased.vcf.gz,http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20201028_3202_phased/CCDG_14151_B01_GRM_WGS_2020-08-05_chr22.filtered.shapeit2-duohmm-phased.vcf.gz.tbi
```
Do you think this is feasible?

Reply:
This could be done but this can't be used for the github CI as it would be too much data to handle I think.
But for the test_full we will for sure use this kind of data !

Reply:
Indeed, we don’t have to include all tests in the GitHub CI environment, especially when dealing with such big data files. 

I tested the medium-sized dataset locally (it took some time to download the files), but the process broke. This suggests we need to further explore what happens with phasing in full chromosomes.

Reply:
I'll try that and fix it next week !